diff --git a/build/parsePreamble.cc b/build/parsePreamble.cc
index c5bd80e..078aa12 100644
--- a/build/parsePreamble.cc
+++ b/build/parsePreamble.cc
@@ -1294,15 +1294,7 @@ int parsePreamble(rpmSpec spec, int initialPackage, enum parseStages stage)
 	}
 
 	if (!spec->buildDir) {
-	    /* Using release here causes a buildid no-recompute test to fail */
-	    char *bn = rpmExpand("%{NAME}-%{VERSION}-build", NULL);
-	    /* Tilde and caret in paths are evil, convert to underscores */
-	    for (char *t = bn; *t; t++) {
-		if (*t == '^' || *t == '~')
-		    *t = '_';
-	    }
-	    spec->buildDir = rpmExpand("%{_top_builddir}/", bn, NULL);
-	    free(bn);
+	    spec->buildDir = rpmExpand("%{_top_builddir}", NULL);
 
 	    /* Override toplevel _builddir for backwards compatibility */
 	    rpmPushMacroFlags(spec->macros, "_builddir", NULL, spec->buildDir,
