--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -416,11 +416,10 @@
 
 # try to ensure some compiler sanity and hardening options where supported
 foreach (flag -fno-strict-overflow -fno-delete-null-pointer-checks -fhardened)
-	check_c_compiler_flag(${flag} found)
-	if (found)
+	check_c_compiler_flag(${flag} compiler-supports${flag})
+	if (compiler-supports${flag})
 		add_compile_options(${flag})
 	endif()
-	unset(found)
 endforeach()
 
 # generated sources
